### PR TITLE
Refactor Person to Student

### LINKED ISSUE #24: Refactor Person to Student

### PR DESCRIPTION
**Pull request by AdityaB4**
Resolves #24.

Apologies for the large changes, there was not much logical segregation or steps to split this task up.

### CODE REVIEW

**Comment by AdityaB4** [comment]
Note: The force pushed commit amends the previous commit to fix a typo while breaking lines longer than 120 chars.

File: src/main/java/seedu/address/logic/Messages.java [hunk 1]
```diff
@@ -14,8 +14,8 @@ public class Messages {
 
     public static final String MESSAGE_UNKNOWN_COMMAND = "Unknown command";
     public static final String MESSAGE_INVALID_COMMAND_FORMAT = "Invalid command format! \n%1$s";
-    public static final String MESSAGE_INVALID_PERSON_DISPLAYED_INDEX = "The person index provided is invalid";
-    public static final String MESSAGE_PERSONS_LISTED_OVERVIEW = "%1$d persons listed!";
+    public static final String MESSAGE_INVALID_PERSON_DISPLAYED_INDEX = "The student index provided is invalid";
+    public static final String MESSAGE_PERSONS_LISTED_OVERVIEW = "%1$d students listed!";
```
Comment (yadunut) [inline src/main/java/seedu/address/logic/Messages.java]: For complete renaming, the PERSON and PERSONS in the variable names can be replaced to student

File: src/main/java/seedu/address/model/Model.java [hunk 2]
```diff
@@ -5,14 +5,14 @@
...
-import seedu.address.model.person.Person;
+import seedu.address.model.student.Student;
 
 /**
  * The API of the Model component.
  */
 public interface Model {
     /** {@code Predicate} that always evaluate to true */
-    Predicate<Person> PREDICATE_SHOW_ALL_PERSONS = unused -> true;
+    Predicate<Student> PREDICATE_SHOW_ALL_PERSONS = unused -> true;
```
Comment (yadunut) [inline src/main/java/seedu/address/model/Model.java]: PERSONS to STUDENTS

**Review by yadunut — CHANGES_REQUESTED** [review]
Great work on this very fat PR. Some nits regarding the naming. There is still exists references to person in the codebase, which can be found by running the command `grep -R --ignore-case PERSON .` from the `tp` directory.

File: src/main/java/seedu/address/model/Model.java (hunk 2 above)
Comment (AdityaB4) [reply src/main/java/seedu/address/model/Model.java]: Thanks for the catch, I might not have realised the find tool I used had case sensitive matching.

**Comment by yadunut** [comment]
Also tests seem to be failing due to not renaming json files, but renaming it in the code, for example, `JsonAddressBookStorageTest.java` in the `readAddressBook_invalidAndValidStudentAddressBook_throwDataLoadingException` test, it references `invalidAndValidStudentAddressBook.json`, but that file has not been renamed from `invalidAndValidPersonAddressBook.json`

**Review by yadunut — APPROVED** [review]
LGTM! Optimally would like another eye on this!

**Comment by yadunut** [comment]
Also for future notices, generally if there is a breaking change, for example in `data/addressbook.json`, a file which isn't committed and local to the developer / user, for larger projects we write a migration script, to migrate the old data to the new one. This is also used in production environments if the project is already deployed, to ensure that the data in database is not lost, but migrated to the new and updated schema. Since this is a smaller project, the updated addressbook.json with seed data can be shared manually among ourselves.

**Review by blaukc — APPROVED** [review]
Looks good

**Pull request merged**